lms/services/dashboard.py

```diff
@@ -246,13 +275,68 @@ def get_course_roster(
                 # For launch data we always add the "active" column as true for compatibility with the roster query.
             ).add_columns(true())
 
+        return roster_last_updated, query.order_by(LMSUser.display_name, LMSUser.id)
+
+    def get_segments_roster(
+        self, segments: list[LMSSegment], h_userids: list[str] | None = None
+    ) -> tuple[datetime | None, Select[tuple[LMSUser, bool]]]:
+        """Return a query that fetches the roster for a segment."""
+        assert (
+            len({segment.lms_course_id for segment in segments}) == 1
+        ), "Segments must belong to the same course"
+
+        rosters_enabled = segments[
+            0
+        ].lms_course.course.application_instance.settings.get("dashboard", "rosters")
+        rosters_last_updated = self._get_segment_rosters_last_updated(segments)
+        # Only use roster data if we have it for all segments, othwrwise it becomes tricky to build a coeherent roster
+        if rosters_enabled and rosters_last_updated:
+            # Use the oldest last updated date as the last updated date for the combined roster
+            roster_last_updated = min(rosters_last_updated)
+            # If rostering is enabled and we do have the data, use it
+            query = self._roster_service.get_segments_roster(
+                segments,
+                role_scope=RoleScope.COURSE,
+                role_type=RoleType.LEARNER,
+                h_userids=h_userids,
+            )
+
+        else:
+            # If we are not going to return data from the roster, don't return the last updated date
+            roster_last_updated = None
```

Comment:
Perhaps we could avoid calling to `self._get_segment_rosters_last_updated(segments)` entirely if `rosters_enabled` is False. Then we wouldn't need to remember unsetting it in the `else` clause.

```suggestion
        rosters_last_updated = self._get_segment_rosters_last_updated(segments) if rosters_enabled else None
        # Only use roster data if we have it for all segments, othwrwise it becomes tricky to build a coeherent roster
        if rosters_enabled and rosters_last_updated:
            # Use the oldest last updated date as the last updated date for the combined roster
            roster_last_updated = min(rosters_last_updated)
            # If rostering is enabled and we do have the data, use it
            query = self._roster_service.get_segments_roster(
                segments,
                role_scope=RoleScope.COURSE,
                role_type=RoleType.LEARNER,
                h_userids=h_userids,
            )

        else:
```

> I made a similar comment in [#6925](https://github.com/hypothesis/lms/pull/6925#discussion_r1914565576), but adding it here for context